api/src/db/destinations.rs

```diff
@@ -102,28 +102,18 @@ pub struct Destination {
     pub config: DestinationConfig,
 }
 
-pub async fn create_destination(
-    pool: &PgPool,
+pub async fn create_destination<'c, E>(
+    executor: E,
     tenant_id: &str,
     name: &str,
     config: DestinationConfig,
     encryption_key: &EncryptionKey,
-) -> Result<i64, DestinationsDbError> {
+) -> Result<i64, DestinationsDbError>
+where
+    E: Executor<'c, Database = Postgres>,
```

Comment:
This bound can be replaced with `E: PgExecutor<'c>` everywhere to make it even shorter. `sqlx::PgExecutor<'c>` is an alias for `Executor<'_, Database = Postgres>`